x-pack/plugins/apm/public/components/shared/redirect_with_offset/index.tsx

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useLocation, Redirect } from 'react-router-dom';
+import qs from 'query-string';
+import React from 'react';
+import { useApmRouter } from '../../../hooks/use_apm_router';
+import { isRouteWithTimeRange } from '../is_route_with_time_range';
+import {
+  TimeRangeComparisonEnum,
+  dayAndWeekBeforeToOffsetMap,
+} from '../../../components/shared/time_comparison/get_comparison_options';
+import { useApmPluginContext } from '../../../context/apm_plugin/use_apm_plugin_context';
+import { getComparisonEnabled } from '../../../components/shared/time_comparison/get_comparison_enabled';
+import { toBoolean } from '../../../context/url_params_context/helpers';
+
+export function RedirectWithOffset({
+  children,
+}: {
+  children: React.ReactElement;
+}) {
+  const { core } = useApmPluginContext();
+  const location = useLocation();
+  const apmRouter = useApmRouter();
+  const matchesRoute = isRouteWithTimeRange({ apmRouter, location });
+  const query = qs.parse(location.search);
+
+  if (
+    matchesRoute &&
+    ('comparisonType' in query || !('comparisonEnabled' in query))
```

Comment:
this is a little hard for me to parse, do you mind leaving a comment why we're redirecting in this scenario? (IIRC, we redirect if comparisonType is set, because we no longer support it, and we redirect if comparisonEnabled is _not_ set, because it is now required)